libpod/rootless_cni_linux.go

```diff
@@ -40,8 +40,12 @@ const (
 //
 // AllocRootlessCNI does not lock c. c should be already locked.
 func AllocRootlessCNI(ctx context.Context, c *Container) (ns.NetNS, []*cnitypes.Result, error) {
-	if len(c.config.Networks) == 0 {
-		return nil, nil, errors.New("allocRootlessCNI shall not be called when len(c.config.Networks) == 0")
+	networks, err := c.networks()
+	if err != nil {
+		return nil, nil, err
+	}
+	if len(networks) == 0 {
+		return nil, nil, errors.New("allocRootlessCNI shall not be called when container has not joined any CNI networks")
```

Comment:
nit to consider if you get other changes,   I'm not a big fan of function names in an error message and "shall" doesn't really fit as the function did do so if it got to here.  Suggest: 
```suggestion
		return nil, nil, errors.New("Allocating a rootless CNI can not be completed when the container has not yet joined any CNI networks")
```

Reply:
I agree

Reply:
Fixed